java/src/jmri/jmrix/loconet/LnReporter.java

```diff
@@ -63,12 +61,13 @@ public int getNumber() {
     }
 
     /**
-      * {@inheritDoc}
+      * Process loconet message handed to us from the LnReporterManager
+      * @param l - a loconetmessage.
       */
-    @Override
     public void message(LocoNetMessage l) {
         // check message type
-        if ((l.getOpCode() == 0xD0) && ((l.getElement(1) & 0xC0) == 0)) {
+        if (((l.getOpCode() == LnConstants.OPC_MULTI_SENSE) && ((l.getElement(1) & 0xC0) == 0))
+                    || (l.getOpCode() == LnConstants.OPC_PEER_XFER && l.getElement(1) == 0x09 && l.getElement(2) == 00 )) {
```

Comment:
This change codes an incomplete decode of OPC_PEER_XFER messages.

Decode of OPC_PEER_XFER must _always_ decoder the **upper nibble** of **<PXCT1>** and the **upper nibble** of **<PXCT2>**.  Failure to decode these upper nibbles _will_ mistakenly allow the LnReporter to interpret other types of LocoNet message as if they were transponding messages.

Reply:
Used the code interpretOpcPeerXfer9(I think you may have written), which just looks at the first 3 bytes to say its a findreply. Skipped the check for truly weird devices, since all we needed was the loco address.

Reply:
Agreed, my mistake.  Forgot that transponding find uses a 9-byte message.  I hope that the decode we use (based on reverse-engineering) is sufficient going forward.